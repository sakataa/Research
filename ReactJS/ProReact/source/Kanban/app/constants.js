export default {
    CARD: 'card',
    FETCH_CARDS: 'fetch cards',
    FETCH_CARDS_SUCCESS: 'fetch cards success',
    FETCH_CARDS_SUCCESS: 'fetch cards error',

    TOGGLE_CARD_DETAILS: 'toggle card details',
    UPDATE_CARD_STATUS: 'update card status',
    UPDATE_CARD_POSITION: 'update card position',

    PERSIST_CARD_DRAG: 'persist card drag',
    PERSIST_CARD_DRAG_SUCCESS: 'persist card drag success',
    PERSIST_CARD_DRAG_ERROR: 'persist card drag error',

    CREATE_TASK: 'create task',
    CREATE_TASK_SUCCESS: 'create task success',
    CREATE_TASK_ERROR: 'create task error',
    DELETE_TASK: 'delete task',
    DELETE_TASK_SUCCESS: 'delete task success',
    DELETE_TASK_ERROR: 'delete task error',
    TOGGLE_TASK: 'toggle task',
    TOGGLE_TASK_SUCCESS: 'toggle task success',
    TOGGLE_TASK_ERROR: 'toggle task error'
};