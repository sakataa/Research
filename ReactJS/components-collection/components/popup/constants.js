export const DEFAULT_WIDTH = '55%';
export const DEFAULT_HEIGHT = '55%';