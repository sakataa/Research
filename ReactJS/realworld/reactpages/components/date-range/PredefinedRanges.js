import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import classnames from 'classnames';
import parseInput from './utils/parseInput.js';
import { defaultClasses } from './styles.js';
import defaultOptions from './defaultOptions.js';

class PredefinedRanges extends Component {

  constructor(props, context) {
    super(props, context);

    this.styles = this.props.theme;
  }

  handleSelect(name, event) {
    event.preventDefault();

    const range = this.props.ranges[name];

    this.props.onSelect({
      startDate: parseInput(range['startDate'], null, 'startOf'),
      endDate: parseInput(range['endDate'], null, 'endOf'),
    });
  }

  clickButton(event, action) {
    event.preventDefault();
    this.props.buttonHandlers[action]();
  }

  renderRangeList(classes) {
    const { ranges, range, onlyClasses } = this.props;
    const { styles } = this;

    return Object.keys(ranges).map(name => {
      const active = (
        parseInput(ranges[name].startDate, null, 'startOf').isSame(range.startDate) &&
        parseInput(ranges[name].endDate, null, 'endOf').isSame(range.endDate)
      );

      const style = {
        ...styles['PredefinedRangesItem'],
        ...(active ? styles['PredefinedRangesItemActive'] : {}),
      };

      const predefinedRangeClass = classnames({
        [classes.predefinedRangesItem]: true,
        [classes.predefinedRangesItemActive]: active
      });

      return (
        <a
          href='#'
          key={'range-' + name}
          className={predefinedRangeClass}
          style={onlyClasses ? undefined : style}
          onClick={(e) => this.handleSelect(name, e)}
        >
          {name}
        </a>
      );
    });
  }

  renderRangeButtons(classes) {
    const { style, onlyClasses, locale } = this.props;
    const locales = { ...defaultOptions.locale, ...locale };
    const { styles } = this;
    const action = {
      apply: "apply",
      cancel: "cancel"
    }

    return (
      <div className={defaultClasses.rangeButtonGroup}>
        <button className={classes.applyButton}
          style={onlyClasses ? undefined : { ...styles['RangeApplyButton'], ...style }}
          type="button"
          onClick={(e) => this.clickButton(e, action.apply)}>
          {locales.applyLabel}
        </button>

        <button className={classes.cancelButton}
          style={onlyClasses ? undefined : { ...styles['RangeCancelButton'], ...style }}
          onClick={(e) => this.clickButton(e, action.cancel)}
          type="button">
          {locales.cancelLabel}
        </button>
      </div>
    )
  }

  render() {
    const { style, onlyClasses, classNames } = this.props;
    const { styles } = this;

    const classes = { ...defaultClasses, ...classNames };

    return (
      <div
        style={onlyClasses ? undefined : { ...styles['PredefinedRanges'], ...style }}
        className={classes.predefinedRanges}
      >
        {this.renderRangeList(classes)}
        {this.renderRangeButtons(classes)}
      </div>
    );
  }
}

PredefinedRanges.defaultProps = {
  onlyClasses: false,
  classNames: {}
};

PredefinedRanges.propTypes = {
  ranges: PropTypes.object.isRequired,
  onlyClasses: PropTypes.bool,
  classNames: PropTypes.object
}

export default PredefinedRanges;
