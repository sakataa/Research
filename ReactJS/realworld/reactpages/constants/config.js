export const debuggingConfig = {
    name: "__lrfConfig__",
    env: "local"
}