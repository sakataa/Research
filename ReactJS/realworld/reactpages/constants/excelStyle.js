export const titleCellStyle = {
    font: { color: { rgb: "FFFFFF" } },
    fill: {
        patternType: "solid",
        fgColor: { rgb: "4C6375" }
    }
}

export const headerCellStyle = {
    font: { bold: true, color: { rgb: "2C2C2C" }, sz: "12" },
    alignment: { horizontal: "center", vertical: "center" },
    fill: { patternType: "solid", fgColor: { rgb: "9EB4C3" } },
    border: { left: { style: "thin", color: { rgb: "D6D6D6" } } }
}

const bodyTextStyle = { alignment: { vertical: "center" } };
const bodyDecimalStyle = { numFmt: "#,##0.00" };
export const bodyCellStyle = {
    rowspan: bodyTextStyle,
    center: { alignment: {...bodyTextStyle.alignment, horizontal: "center" } },
    number: { numFmt: "#,##0" },
    decimal: bodyDecimalStyle,
    negativeDecimal: {...bodyDecimalStyle, font: { color: { rgb: "FF0000" } } }
}

const totalTextStyle = { font: { bold: true } };
const totalDecimalStyle = {...totalTextStyle, numFmt: "#,##0.00" };
export const totalCellStyle = {
    default: totalTextStyle,
    number: {...totalTextStyle, numFmt: "#,##0" },
    decimal: totalDecimalStyle,
    negativeDecimal: {...totalDecimalStyle, font: { bold: true, color: { rgb: "FF0000" } } }
}

const grandTotalTextStyle = {
    font: { bold: true },
    fill: {
        patternType: "solid",
        fgColor: { rgb: "B8C3C9" }
    }
};
const decimalGrandTotalStyle = {...grandTotalTextStyle, numFmt: "#,##0.00" };
export const grandTotalCellStyle = {
    default: grandTotalTextStyle,
    number: {...grandTotalTextStyle, numFmt: "#,##0" },
    decimal: decimalGrandTotalStyle,
    negativeDecimal: {...decimalGrandTotalStyle, font: { bold: true, color: { rgb: "FF0000" } } }
}