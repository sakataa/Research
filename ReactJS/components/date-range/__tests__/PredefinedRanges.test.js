import React from 'react'
import renderer from 'react-test-renderer'

import moment from 'moment'
import { MomentFormat } from '../constants'
import getTheme from '../styles'
import locale from "../utils/locale"
import PredefinedRanges from '../PredefinedRanges'


describe('PredefinedRanges', () => {
    const currentDate = moment("11/23/2017", MomentFormat.default);
    const yesterday = {
        startDate: moment("11/22/2017", MomentFormat.default),
        endDate: moment("11/22/2017", MomentFormat.default)
    };
    const lastMonth = {
        startDate: moment("10/1/2017", MomentFormat.default),
        endDate: moment("10/30/2017", MomentFormat.default)
    };
    const defaultProps = {
        ranges: { 'Yesterday': yesterday, 'LastMonth': lastMonth },
        onlyClasses: false,
        classNames: {}
    }
    const theme = getTheme({ Day: { margin: 5 } });

    afterAll(() => {
        jest.resetAllMocks();
    })

    describe('constructor', () => {
        it('initial value should be correct', () => {
            const componentInstance = new PredefinedRanges(defaultProps);

            expect(componentInstance.styles).toBe(undefined);
            expect(componentInstance.selectedItem).toBe(null);
        })
    })

    describe('handleSelect', () => {
        it('event.preventDefault() should be called', () => {
            const event = {
                preventDefault: jest.fn()
            }
            const onSelect = jest.fn();
            const newProps = { onSelect };
            const props = Object.assign({}, defaultProps, newProps);
            const componentInstance = new PredefinedRanges(props);

            const rangeName = "Yesterday";
            componentInstance.handleSelect(rangeName, event);

            expect(event.preventDefault).toHaveBeenCalled();
        })

        it('onSelect props should be called with correct parameter', () => {
            const expectedParam = defaultProps.ranges.Yesterday;
            const event = {
                preventDefault: jest.fn()
            }
            const onSelect = jest.fn();
            const newProps = { onSelect };
            const props = Object.assign({}, defaultProps, newProps);
            const componentInstance = new PredefinedRanges(props);

            const rangeName = "Yesterday";
            componentInstance.handleSelect(rangeName, event);

            expect(onSelect).toHaveBeenCalledWith(expectedParam);
        })
    })

    describe('clickButton', () => {
        it('event.preventDefault() should be called', () => {
            const buttonHandlers = {
                "Apply": jest.fn(),
                "Cancel": jest.fn()
            };
            const event = {
                preventDefault: jest.fn()
            }
            const newProps = { buttonHandlers };
            const props = Object.assign({}, defaultProps, newProps);
            const componentInstance = new PredefinedRanges(props);
            const actionName = "Apply";

            componentInstance.clickButton(event, actionName);

            expect(event.preventDefault).toHaveBeenCalled();
        })

        it('event.preventDefault() should be called', () => {
            const buttonHandlers = {
                "Apply": jest.fn(),
                "Cancel": jest.fn()
            };
            const event = {
                preventDefault: jest.fn()
            }
            const newProps = { buttonHandlers };
            const props = Object.assign({}, defaultProps, newProps);
            const componentInstance = new PredefinedRanges(props);
            const actionName = "Apply";

            componentInstance.clickButton(event, actionName);

            expect(buttonHandlers.Apply).toHaveBeenCalled();
        })
    })

    describe('renderRangeList', () => {
        const classes = {
            predefinedRangesItem: "predefinedRangesItem",
            predefinedRangesItemActive: "predefinedRangesItemActive"
        }
        describe('Assert props', () => {
            function assertByOnlyClassesProp({ onlyClasses, expectedStyle }) {
                it(`onlyClasses = ${onlyClasses.toString()} - range element has correct props`, () => {
                    const expectedResult = {
                        style: expectedStyle
                    }
                    const newProps = {
                        onlyClasses,
                        theme
                    }

                    const props = Object.assign({}, defaultProps, newProps);
                    const componentInstance = new PredefinedRanges(props);
                    componentInstance.selectedItem = "Yesterday";
                    const ranges = componentInstance.renderRangeList(classes);

                    const result = ranges[0].props;

                    expect(result.style).toEqual(expectedResult.style);
                })
            }

            const style = {
                ...theme.PredefinedRangesItem,
                ...theme.PredefinedRangesItemActive
            }
            const testList = [
                { onlyClasses: true, expectedStyle: undefined },
                { onlyClasses: false, expectedStyle: style }
            ]

            testList.forEach(assertByOnlyClassesProp);
        })

        it('Yesterday range is not selected - this range should have correct class name', () => {
            const expectedClassName = classes.predefinedRangesItem;
            const newProps = {
                onlyClasses: true,
                theme
            }
            const props = Object.assign({}, defaultProps, newProps);
            const componentInstance = new PredefinedRanges(props);

            const ranges = componentInstance.renderRangeList(classes);

            const yesterdayRange = ranges[0];
            const result = yesterdayRange.props.className;

            expect(result).toBe(expectedClassName);
        })

        it('Yesterday range is selected - this range should have correct class name', () => {
            const expectedClassName = Object.keys(classes).map(item => item).join(" ");
            const newProps = {
                onlyClasses: true,
                theme
            }
            const props = Object.assign({}, defaultProps, newProps);
            const componentInstance = new PredefinedRanges(props);
            componentInstance.selectedItem = "Yesterday";

            const ranges = componentInstance.renderRangeList(classes);

            const yesterdayRange = ranges[0];
            const result = yesterdayRange.props.className;

            expect(result).toBe(expectedClassName);
        })

        it('onClick event - onSelect should be called with correct parameter', () => {
            const newProps = {
                onlyClasses: true,
                theme
            }
            const props = Object.assign({}, defaultProps, newProps);
            const componentInstance = new PredefinedRanges(props);
            componentInstance.handleSelect = jest.fn();

            const ranges = componentInstance.renderRangeList(classes);

            const event = {},
                rangeName = "Yesterday";
            const yesterdayRange = ranges[0];

            yesterdayRange.props.onClick(event, rangeName);

            expect(componentInstance.handleSelect).toHaveBeenCalled();
        })
    })

    describe('renderRangeButtons', () => {
        const classes = {
            applyButton: "applyButton",
            cancelButton: "cancelButton"
        }
        const defaultPropsOfRenderRangeButtons = {
            onlyClasses: true,
            theme,
            locale
        }

        function getDefaultRangeButtons(newProps = {}) {
            const props = Object.assign({}, defaultProps, defaultPropsOfRenderRangeButtons, newProps);
            const componentInstance = new PredefinedRanges(props);
            return componentInstance.renderRangeButtons(classes);
        }

        it('class name of container should be correct', () => {
            const rangeButtons = getDefaultRangeButtons();
            const containerClassName = rangeButtons.props.className;
            const expectedContainerClassName = "range-button-group";

            expect(containerClassName).toBe(expectedContainerClassName);
        })

        it('class name of buttons should be correct', () => {
            const rangeButtons = getDefaultRangeButtons();
            const buttons = rangeButtons.props.children;

            const applyButtonProps = buttons[0].props;
            const expectedClassNameOfApplyButton = classes.applyButton;

            const cancelButtonProps = buttons[1].props;
            const expectedClassNameOfCancelButton = classes.cancelButton;

            expect(applyButtonProps.className).toBe(expectedClassNameOfApplyButton);
            expect(cancelButtonProps.className).toBe(expectedClassNameOfCancelButton);
        })

        it('buttons text should be correct', () => {
            const applyLabel = "Apply",
                cancelLabel = "Cancel";
            const newProps = {
                locale: { applyLabel, cancelLabel }
            }
            const rangeButtons = getDefaultRangeButtons(newProps);
            const buttons = rangeButtons.props.children;

            const applyButtonProps = buttons[0].props;
            const cancelButtonProps = buttons[1].props;

            expect(applyButtonProps.children).toBe(applyLabel);
            expect(cancelButtonProps.children).toBe(cancelLabel);
        })

        it('buttons onClick event - clickButton should be called with correct parameter', () => {
            const applyActionName = "apply",
                cancelActionName = "cancel";

            const props = Object.assign({}, defaultProps, defaultPropsOfRenderRangeButtons);
            const componentInstance = new PredefinedRanges(props);
            componentInstance.clickButton = jest.fn();
            const rangeButtons = componentInstance.renderRangeButtons(classes);
            const buttons = rangeButtons.props.children;

            const applyButtonProps = buttons[0].props;
            const cancelButtonProps = buttons[1].props;

            const event = {};
            applyButtonProps.onClick(event);
            expect(componentInstance.clickButton).toHaveBeenCalledWith(event, applyActionName);

            cancelButtonProps.onClick(event);
            expect(componentInstance.clickButton).toHaveBeenCalledWith(event, cancelActionName);
        })

        describe('Assert style', () => {
            function assertByOnlyClassesProp({ onlyClasses, expectedStyle, buttonIndex }) {
                it(`onlyClasses = ${onlyClasses.toString()} - button element has correct props`, () => {
                    const expectedResult = {
                        style: expectedStyle
                    }
                    const newProps = {
                        onlyClasses,
                        theme
                    }

                    const props = Object.assign({}, defaultProps, newProps);
                    const componentInstance = new PredefinedRanges(props);
                    const rangeButtons = componentInstance.renderRangeButtons(classes);
                    const buttons = rangeButtons.props.children;

                    const buttonProps = buttons[buttonIndex].props;

                    expect(buttonProps.style).toEqual(expectedResult.style);
                })
            }

            const testList = [
                { onlyClasses: true, expectedStyle: undefined, buttonIndex: 0 },
                { onlyClasses: false, expectedStyle: theme.RangeApplyButton, buttonIndex: 0 },
                { onlyClasses: true, expectedStyle: undefined, buttonIndex: 1 },
                { onlyClasses: false, expectedStyle: theme.RangeCancelButton, buttonIndex: 1 }
            ]

            testList.forEach(assertByOnlyClassesProp);
        })
    })

    describe('render', () => {
        const classNames = {
            predefinedRanges: "predefinedRanges"
        }
        const defaultPropsOfRender = {
            onlyClasses: true,
            theme,
            classNames
        }

        it('onlyClasses = true - match snapshot', () => {
            const props = Object.assign({}, defaultProps, defaultPropsOfRender);
            const component = renderer.create(<PredefinedRanges {...props} />);

            const tree = component.toJSON();

            expect(tree).toMatchSnapshot();
        })

        it('onlyClasses = false - match snapshot', () => {
            const newProps = { onlyClasses: false };
            const props = Object.assign({}, defaultProps, defaultPropsOfRender, newProps);
            const component = renderer.create(<PredefinedRanges {...props} />);

            const tree = component.toJSON();

            expect(tree).toMatchSnapshot();
        })
    })
});