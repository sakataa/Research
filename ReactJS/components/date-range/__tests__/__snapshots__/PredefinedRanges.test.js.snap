// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredefinedRanges render onlyClasses = false - match snapshot 1`] = `
<div
  className="predefinedRanges"
  style={
    Object {
      "display": "inline-block",
      "marginLeft": 10,
      "marginTop": 10,
      "verticalAlign": "top",
      "width": 160,
    }
  }
>
  <a
    className="rdr-PredefinedRangesItem"
    href="#"
    onClick={[Function]}
    style={
      Object {
        "borderRadius": "4px",
        "color": "#2c3e50",
        "display": "block",
        "fontSize": 12,
        "marginBottom": 3,
        "padding": "10px 14px",
        "textDecoration": "none",
      }
    }
  >
    Yesterday
  </a>
  <a
    className="rdr-PredefinedRangesItem"
    href="#"
    onClick={[Function]}
    style={
      Object {
        "borderRadius": "4px",
        "color": "#2c3e50",
        "display": "block",
        "fontSize": 12,
        "marginBottom": 3,
        "padding": "10px 14px",
        "textDecoration": "none",
      }
    }
  >
    LastMonth
  </a>
  <div
    className="range-button-group"
  >
    <button
      className="rdr-Button rdr-ApplyButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f05b1f",
          "border": "none",
          "borderRadius": "2px",
          "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)",
          "color": "#fff",
          "display": "inline-block",
          "height": 25,
          "letterSpacing": "0.5px",
          "lineHeight": "25px",
          "marginRight": "calc(4%)",
          "outline": 0,
          "textAlign": "center",
          "verticalAlign": "middle",
          "width": "48%",
        }
      }
      type="button"
    >
      OK
    </button>
    <button
      className="rdr-Button rdr-CancelButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#757985",
          "border": "none",
          "borderRadius": "2px",
          "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)",
          "color": "#fff",
          "display": "inline-block",
          "height": 25,
          "letterSpacing": "0.5px",
          "lineHeight": "25px",
          "outline": 0,
          "textAlign": "center",
          "verticalAlign": "middle",
          "width": "48%",
        }
      }
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`PredefinedRanges render onlyClasses = true - match snapshot 1`] = `
<div
  className="predefinedRanges"
  style={undefined}
>
  <a
    className="rdr-PredefinedRangesItem"
    href="#"
    onClick={[Function]}
    style={undefined}
  >
    Yesterday
  </a>
  <a
    className="rdr-PredefinedRangesItem"
    href="#"
    onClick={[Function]}
    style={undefined}
  >
    LastMonth
  </a>
  <div
    className="range-button-group"
  >
    <button
      className="rdr-Button rdr-ApplyButton"
      onClick={[Function]}
      style={undefined}
      type="button"
    >
      OK
    </button>
    <button
      className="rdr-Button rdr-CancelButton"
      onClick={[Function]}
      style={undefined}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;
